swagger-backend.yaml

```diff
@@ -0,0 +1,191 @@
+  - application/json
+produces:
+  - application/json
+basePath: /api/flotta-backend/v1
+definitions:
+  error:
+    type: object
+    required:
+      - message
+    properties:
+      message:
+        type: string
+  device-configuration-response:
+    type: object
+    properties:
+      message:
+        type: string
+      device-configuration:
+        $ref: 'swagger.yaml#/definitions/device-configuration'
+  device-registration-request:
+    type: object
+    properties:
+      status:
+        type: boolean
+  device-registration-response:
+    type: object
+    properties:
+      target-namespace:
+        type: string
+      message:
+        type: string
+paths:
+  '/namespaces/{namespace}/devices/{device-id}/deregister':
+    get:
+      description: Confirms if a device can be deregistered
+      operationId: IsDeviceDeregistrable
+      parameters:
+        - in: path
+          name: namespace
+          description: Namespace where the device resides
+          type: string
+          required: true
+        - in: path
+          name: device-id
+          description: Device ID
+          type: string
+          required: true
+      responses:
+        '200':
+          description: Success
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/error'
+  '/namespaces/{namespace}/devices/{device-id}/configuration':
+    get:
+      description: Returns the device configuration
+      operationId: GetDeviceConfiguration
+      parameters:
+        - in: path
+          name: namespace
+          description: Namespace where the device resides
+          type: string
+          required: true
+        - in: path
+          name: device-id
+          description: Device ID
+          type: string
+          required: true
+      responses:
+        '200':
+          description: Success
+          schema:
+            $ref: '#/definitions/device-configuration-response'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/device-configuration-response'
+  '/namespaces/{namespace}/devices/{device-id}/enrol':
+    post:
+      description: Initiates the process of enrolling the device
+      operationId: EnrolDevice
+      parameters:
+        - in: path
+          name: namespace
+          description: Namespace where the device resides
+          type: string
+          required: true
+        - in: path
+          name: device-id
+          description: Device ID
+          type: string
+          required: true
+        - in: body
+          name: enrolment-info
+          required: true
+          schema:
+            $ref: 'swagger.yaml#/definitions/enrolment-info'
+      responses:
+        '200':
+          description: Success
```

Comment:
We need to somehow encode in the response that the device is already present - either in the body or status code. Maybe HTTP 201 (Created) when new enrollment is recoded and HTTP 200 (OK) when the device already exists?

Reply:
Sounds good. Thanks for the suggestion.